File: core/src/main/java/javabot/operations/time/CountryParser.java
package javabot.operations.time;

public class CountryParser {
    private static Tri<Country> tri = new Tri<Country>( );

    static {
        for( Country country : Country.values( ) ) {
            tri.insert( country.getName( ), country );
            tri.insert( country.getAbbreviation( ), country );
        }
    }

    public Country getCountry( String key ) {
        return tri.get( key );
    }
}

File: src/main/kotlin/javabot/operations/time/CountryParser.kt
package javabot.operations.time

public class CountryParser {

    public fun getCountry(key: String): Country {
        return tri.get(key)
    }

    companion object {
        private val tri = Tri<Country>()

        init {
            for (country in Country.values()) {
                tri.insert(country.name, country)
                tri.insert(country.abbreviation, country)
            }
        }
    }
}
